src/main/java/org/interledger/ilqp/QuoteByDestinationAmountRequest.java

```diff
@@ -67,82 +66,81 @@ public Builder destinationHoldDuration(final Duration destinationHoldDuration) {
       this.destinationHoldDuration = Objects.requireNonNull(destinationHoldDuration);
       return this;
     }
-    
+
     /**
      * The method that actually constructs a QuoteByDestinationAmountRequest.
-     * 
+     *
      * @return An instance of {@link QuoteByDestinationAmountRequest}
      */
     public QuoteByDestinationAmountRequest build() {
       return new Builder.Impl(this);
     }
-    
+
     /**
      * A private, immutable implementation of {@link QuoteByDestinationAmountRequest}.
      */
     private static class Impl implements QuoteByDestinationAmountRequest {
 
       private final InterledgerAddress destinationAccount;
+      private final BigInteger destinationAmount;
       private final Duration destinationHoldDuration;
-      private long destinationAmount;
-      
+
       /**
        * Constructs an instance from the values held in the builder.
-       * 
+       *
        * @param builder A Builder used to construct {@link QuoteByDestinationAmountRequest}
-       *        instances.
+       *                instances.
        */
       private Impl(final Builder builder) {
         Objects.requireNonNull(builder);
-        
+
         this.destinationAccount = Objects.requireNonNull(builder.destinationAccount,
             "destinationAccount must not be null!");
-        
-        if (builder.destinationAmount < 0) {
```

Comment:
is there a compelling reason to remove the < 0 check? We cant put negative numbers on the wire, so it seems like a safe thing to do?

Reply:
Per our discussion in the Java conference call, this has been restored.